src/main/java/com/google/sps/util/SearchUtils.java

```diff
@@ -0,0 +1,98 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.util;
+
+import com.google.appengine.api.datastore.Query;
+import java.lang.UnsupportedOperationException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * A utility class with static methods to complete search related functions.
+ */
+public class SearchUtils {
+  /**
+   * Takes a string and returns a standard version of the string depending on the implementation.
+   * This is used for processing a string before parsing for different query data.
+   * Current implementation returns strings trimmed and in lowercase.
+   */
+  public static String standardize(String str) {
+   String standard = str.trim();
+   standard = standard.toLowerCase();
+   return standard;
+  }
+
+  /*
+   * Takes a string with a user query and parses it looking for "@" symbols to extract author names. The 
+   * result is an arraylist of strings which are the authors the user searched for, without "@" symbol.
+   * TODO: Implement this function with regex.
+   */
+  public static ArrayList<String> parseAuthor(String str) throws 
+    UnsupportedOperationException {
+    throw new UnsupportedOperationException();
+  }
+
+  /*
+   * Takes a string with a user query and parses it looking for "#" symbols to extract tags. The 
+   * result is an arraylist of strings which are the tags the user searched for, without "#" symbol.
+   * TODO: Implement this function when tag system has been decided on
+   */
+  public static ArrayList<String> parseTags(String str) throws 
+    UnsupportedOperationException {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Takes a string of the original query and extracts the plain text with no "@" or "#" symbol,
+   * which therefore represents the user's query for a recipe name.
+   * TODO: Implement regex functionality to not simply return the string passed through.
+   */
+  public static String parseRecipeName(String str) {
+    return str;
+  }
+
+  /**
+   * Takes in an authorID and returns the query filter for the given author so it can be added to
+   * a query.
+   */
+  public static Query.FilterPredicate getAuthorFilter(Long authorID) {
+      return new Query.FilterPredicate("authorID", Query.FilterOperator.EQUAL, authorID);
+  }
+
+  /**
+   * Takes in an authorID and returns the query filter for the given author so it can be added to
+   * a query.
+   */
+  public static Query.FilterPredicate getRecipeNameFilter(String recipeName) {
+      return new Query.FilterPredicate("name", Query.FilterOperator.EQUAL, recipeName);
+  }
```

Comment:
Remember to account for standardized recipeName, versus non-standardized "name" property

Reply:
eventually, anyway.

Reply:
How do you mean?

Reply:
If I understand correctly, recipeName is converted to all lower-case, trimmed, cleaned of special chars, etc. Is the same done for "name" property before they are compared?

Reply:
So I talked to isabel about this, and mentioned the basic cleaning process with her regarding tags and such, but I'll go over it with her to make sure we're on the same page.